### PR TITLE
Nils cache

### PR DESCRIPTION
**Pull request by Didgy74**
Unfinished atm.

Need to merge commits that are in main.

Need to comment code properly. Some comments may have been lost from previous code, will have to go over it.

### CODE REVIEW

**Comment by Didgy74** [comment]
Fixes have been made. Ready for review.

**Review by cecilianor — CHANGES_REQUESTED** [review]

File: lib/Utilities.h [hunk 1]
```diff
@@ -123,4 +125,94 @@ struct ParsedLink {
     ResultType resultType;
 };
 
+namespace Bach {
```
[the remark this answers is not in the document]
Comment (Didgy74) [reply lib/Utilities.h:128]: This is not an issue, this is normal in C++.

File: lib/TileLoader.h [hunk 2]
```diff
@@ -5,40 +5,311 @@
...
+        const QByteArray &byteArray,
+        TileLoadedCallbackFn signalFn);
 };
 
 namespace Bach {
-    QString setPbfLink(const TileCoord &tileCoord, const QString &pbfLinkTemplate);
+    QString setPbfLink(TileCoord tileCoord, const QString &pbfLinkTemplate);
+
+    bool writeTileToDiskCache(
+        const QString& basePath,
```
[the remark this answers is not in the document]
Comment (Didgy74) [reply lib/TileLoader.h:207]: Right, this is just a helper function used by TileLoader AND the unit tests. The unit tests can use it to pre-load a tile-cache in a specific directory, and then be certain the TileLoader will read the cache correctly. It's used internally by TileLoader and from unit tests. The comments have been updated to reflect this.

**Review by EimenOueslati — APPROVED** [review]
Good!

**Pull request closed without merging**